### ISSUE TITLE
Handling of meeting invitations

### ISSUE THREAD

**Issue by jobisoft** [jobisoft/EAS-4-TbSync#10]
_From @Nico321 on February 5, 2018 13:49_

Hi,

Im missing the description of some of the meetings. It appears, that I can see the description for meetings which I created myself, but im missing the notes for meetings, which i got invited to.

This is what a meeting looks like, which I was invited to:
![image](https://user-images.githubusercontent.com/6123294/35807483-39689b1e-0a83-11e8-907c-f2830e07983b.png)

This is an example of a meeting, which I created myself (notice the tabs for description, attachments and participants):
![image](https://user-images.githubusercontent.com/6123294/35807421-03dea63c-0a83-11e8-99f5-de4b85f1fc10.png)

Im using Thunderbird 52.6.0 (64-Bit) on macOS High Sierra 10.13.3. Descriptions are available for all meetings when using CalDav or ActiveSync on iOs.

If you need any further information, feel free to contact me.

_Copied from original issue: jobisoft/TbSync#46_

**Comment by jobisoft** [comment]
I think this is a limitation of Thunderbird. The Invitation dialog does not display the description.

But handling invitations itself (accepting, declining,..) is not yet fully supported by TbSync. I keep this issue open until I made progress on this.

**Comment by jobisoft** [comment]
_From @mdwy62 on May 8, 2018 16:39_

When I try to set up a meeting and invite attendees, I am asked to send emails and I select yes. Everything looks fine on my end, the event is added to my calendar, but no invitees receive the email invitations. I am using Thunderbird 52.7.0 on Ubuntu 16.04. Is the ability to invite others over EAS supposed to work or is it not supported currently? Thanks.

**Comment by jobisoft** [comment]
Handling invitations (accepting, declining,..) is not yet fully supported by TbSync. On some systems it works, on some it does not. I have not yet found the reason. This might take some time...

**Comment by jobisoft** [comment]
_From @crbble on July 6, 2018 8:21_

In my experience TbSync calendars behave like local calendars, which is also why Thunderbird asks to send E-Mails to the attendees (it does not do this for CalDAV events). If this does not work, it is a general local Thunderbird problem (like your hint about the calendar and e-mail address association, which may be checked in the calendar settings). 

All other behaviour will be ruled out by the server and most likely all servers will behave differently. Horde has been messing around with this using CalDAV since several years (for example bugs 3965, 13664 in the horde bugtracker). Problems are the changing of the event owner, modification of the event on every acceptance and others.

**Comment by jobisoft** [comment]
That is correct, I have not yet created my own calendar type, but use the "local calendar type". To fix this issue I eventually have to create my own type so I can control how these emails are send. 

But I will start working on this only after all other non-calendar issues are solved and TbSync is stable.

**Comment by shmerl** [comment]
Will this also handle looking up locations? Right now, when creating a meeting (calendar is using tbsync), it can look up attendees, but it doens't look up any locations.

**Comment by jobisoft** [comment]
I do not think ActiveSync 14 can do that. This is more like something for EWS. But I will check when I start working on this

**Comment by shmerl** [comment]
Yeah, it could be a [separate thing](https://docs.microsoft.com/en-us/exchange/architecture/client-access/availability-service?view=exchserver-2019).

**Comment by btaroli** [comment]
You may begin getting **a lot** of inquiries about this soon. A large employer has shifted it's users to Exchange and there is a fairly large contingent of Thunderbird users. We're trying this add-on to save us from Outlook and it seems that it has some (known) issues handling event invitations (sending and receiving).

Any chance we could wake the calendar dragon?

**Comment by jobisoft** [comment]
Closing this one, eight years on. It was kept open as the umbrella for invitation handling, and that has landed in v5.2.3.

Answering an invitation reaches the server and the organiser (#8, #66). It no longer duplicates the meeting or re-invites the other attendees (#320). The invitation mail and the calendar entry are recognised as the same meeting (#40). An invitation you have not answered no longer claims to be accepted (#4). Inviting others to a meeting you create works too (#7, #25). And the description of a meeting you were invited to syncs like any other, which is where this thread started.

Out now in v5.2.3. If Thunderbird has not offered you the update yet, you find the new versions here: [TbSync](https://github.com/jobisoft/TbSync/releases/tag/v5.2.3), [EAS](https://github.com/jobisoft/EAS-4-TbSync/releases/tag/v5.2.3), [Google](https://github.com/jobisoft/google-4-tbsync/releases/tag/v5.2.3). Update all three together. Accounts set up before v5.2.2 have to be disconnected and connected again before they sync at all - back up pending data first, disconnecting deletes the account's local calendars and address books.
